operatorcourier/manifest_parser.py

```diff
@@ -0,0 +1,82 @@
+import os
+import logging
+from operatorcourier.errors import OpCourierBadBundle
+from operatorcourier import identify
+
+
+logger = logging.getLogger(__name__)
+CRD_KEY = 'CustomResourceDefinition'
+CSV_KEY = 'ClusterServiceVersion'
+
+
+def is_manifest_folder(folder_path):
+    """
+    :param folder_path: the path of the input folder
+    :return: True if the folder contains valid operator manifest files (at least
+             1 valid CSV file), False otherwise
+    """
+    for item in os.listdir(folder_path):
+        item_path = os.path.join(folder_path, item)
+        if not os.path.isfile(item_path):
+            continue
+        if item.endswith(".yaml") or item.endswith(".yml"):
+            with open(item_path) as f:
+                file_content = f.read()
+            if CSV_KEY == identify.get_operator_artifact_type(file_content):
+                return True
+
+    folder_name = os.path.basename(folder_path)
+    logging.warning('Ignoring folder "%s" as it is not a valid manifest '
+                    'folder', folder_name)
+    return False
+
+
+def get_crd_csv_files_content(folder_path):
+    """
+    Given a folder path, the method returns a dict containing manifest files
+    content grouped by CRD and CSV key.
+    :param folder_path: the path of the input folder
+    :return: a dict where the key is either  'ClusterServiceVersion' or
+             'CustomResourceDefinition', and the value is a list of the valid
+             CSV and CRD files content
+    """
+    crd_csv_files_content = {}
+    for item in os.listdir(folder_path):
+        item_path = os.path.join(folder_path, item)
+        if not os.path.isfile(item_path):
+            continue
+        if item.endswith(".yaml") or item.endswith(".yml"):
```

Comment:
This code repeats everywhere, would be nice to factor it out into separate function `is_yaml_file`